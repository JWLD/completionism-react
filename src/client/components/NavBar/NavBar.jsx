import React from 'react';
import { Link } from 'react-router-dom';
import FaHome from 'react-icons/lib/fa/home';

import './NavBar.scss';

const NavBar = () => (
	<nav className="main-nav">
		<Link to="/">
			<FaHome />
		</Link>
	</nav>
);

export default NavBar;
