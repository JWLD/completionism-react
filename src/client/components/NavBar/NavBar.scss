@import 'global';

$navBarPadding: 1.25rem;

.main-nav {
	position: fixed;
	top: 0;
	left: 0;
	z-index: 1;

	width: 100vw;
	height: $navBarHeight;
	padding: $navBarPadding;

	background-color: #373e48;
	box-shadow: 0 0 1.5rem .1rem black;

	.nav-left {
		position: absolute;
		top: 0;
		left: 0;

		padding: $navBarPadding;

		.nav-link:not(:last-child) {
			margin-right: $navBarPadding;
		}
	}

	.nav-link {
		height: $navBarHeight - ($navBarPadding * 2);
		min-width: $navBarHeight - ($navBarPadding * 2);
		padding: .75rem;

		color: white;
		background-color: #21252b;
		border-radius: .4rem;
		font: 2.5rem DinMC;
		text-transform: uppercase;

		svg {
			height: 100%;
			width: 100%;
		}

		&.cat {
			padding: .75rem 1rem;
		}
	}

	.category-nav {
		width: 45rem;
		justify-content: space-between;

		h1 {
			color: white;
			font: 5rem DinMC;
			text-transform: uppercase;
		}
	}
}
