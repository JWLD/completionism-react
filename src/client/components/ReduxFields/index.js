export { default as InputField } from './InputField';
export { default as SelectBoxField } from './SelectBoxField';
