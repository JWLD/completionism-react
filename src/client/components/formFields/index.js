export { default as CheckboxField } from 'formFields/CheckboxField'
export { default as InputField } from 'formFields/InputField'
export { default as SelectBoxField } from 'formFields/SelectBoxField'
