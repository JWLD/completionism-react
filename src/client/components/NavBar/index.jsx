import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link, withRouter } from 'react-router-dom'

import { CONTENT } from 'constants/content'
import * as SC from './styled'

class NavBar extends Component {
  renderContentNav() {
    if (!this.props.match.params.category) return null

    const { category, content } = this.props.match.params
    const prevPage = Number(content) - 1
    const nextPage = Number(content) + 1

    return (
      <SC.CentreWrap>
        <SC.NavLink
          to={`/browse/${category}/${prevPage}`}
          className="fa fa-arrow-circle-left"
        />

        <SC.MainTitle>{CONTENT[content].content}</SC.MainTitle>

        <SC.NavLink
          to={`/browse/${category}/${nextPage}`}
          className="fa fa-arrow-circle-right"
        />
      </SC.CentreWrap>
    )
  }

  renderCategorySpan() {
    const { category } = this.props.match.params

    return category && <SC.NavSpan>{category}</SC.NavSpan>
  }

  render() {
    return (
      <SC.NavBar>
        <SC.LeftWrap>
          <SC.NavLink className="fa fa-home" to="/" />

          {this.renderCategorySpan()}
        </SC.LeftWrap>

        {this.renderContentNav()}
      </SC.NavBar>
    )
  }
}

NavBar.propTypes = {
  match: PropTypes.object.isRequired
}

export default withRouter(NavBar)
