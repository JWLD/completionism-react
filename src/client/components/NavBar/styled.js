import styled from 'styled-components'
import { Link } from 'react-router-dom'
import {
  FaHome,
  FaArrowCircleLeft,
  FaArrowCircleRight
} from 'react-icons/lib/fa'

import { mixins, vars } from 'style'

const navBarPadding = 1.25
const linkSize = vars.navBarHeight - navBarPadding * 2
const iconSize = 3.5

export const NavBar = styled.nav`
  ${mixins.flex};

  z-index: ${vars.navBar};
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: ${vars.navBarHeight}rem;
  background-color: ${vars.mainLight};
  box-shadow: 0 0 1.5rem 0.1rem black;
`

export const LeftWrap = styled.div`
  ${mixins.flex};

  position: absolute;
  top: 0;
  left: 0;
  padding: ${navBarPadding}rem;
`

export const CentreWrap = styled.div`
  ${mixins.flex};

  width: 45rem;
  justify-content: space-between;
`

export const MainTitle = styled.h1`
  color: white;
  font: 5rem DinMC;
  text-transform: uppercase;
`

export const NavLink = styled(Link)`
  ${mixins.flex};

  height: ${linkSize}rem;
  min-width: ${linkSize}rem;
  padding: 0.75rem;
  border-radius: 0.4rem;
  color: white;
  background-color: ${vars.mainDark};
  font-size: 3.25rem;

  :not(:last-child) {
    margin-right: ${navBarPadding}rem;
  }
`

export const NavSpan = NavLink.withComponent('span').extend`
  padding: 0.75rem 1rem;
  font: 2.5rem DinMC;
  text-transform: uppercase;
`

export const HomeIcon = styled(FaHome)`
  height: ${iconSize}rem;
  width: ${iconSize}rem;
`

export const CircleLeftIcon = HomeIcon.withComponent(FaArrowCircleLeft)

export const CircleRightIcon = HomeIcon.withComponent(FaArrowCircleRight)
