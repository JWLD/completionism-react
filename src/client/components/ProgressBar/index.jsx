import React from 'react'
import PropTypes from 'prop-types'

import * as SC from 'ProgressBar/styled'

const ProgressBar = ({ count, height }) => (
  <SC.ProgressBar height={height}>
    <SC.BarFill fill={count} height={height} />
  </SC.ProgressBar>
)

ProgressBar.propTypes = {
  count: PropTypes.number.isRequired,
  height: PropTypes.number
}

ProgressBar.defaultProps = {
  height: 1.5
}

export default ProgressBar
