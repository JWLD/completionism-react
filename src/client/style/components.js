import styled from 'styled-components'

import mixins from 'style/mixins'
import vars from 'style/variables'

export const PageWrap = styled.div`
  ${mixins.flexVertical};

  min-width: 100vw;
  margin-top: ${vars.navBarHeight}rem;
  padding: ${vars.pagePadding}rem;
`

export const BrowseBlock = styled.div`
  ${mixins.flexVertical};

  position: relative;
  width: 100%;
  padding: ${vars.itemListPadding}rem;
  border-radius: 0.4rem;
  background-color: ${vars.mainGrey};

  &:not(:last-child) {
    margin-bottom: 2rem;
  }
`
