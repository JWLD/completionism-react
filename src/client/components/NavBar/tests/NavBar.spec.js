import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { NavBar } from 'components/NavBar'

Enzyme.configure({ adapter: new Adapter() })

describe('<NavBar />', () => {
  let mockProps = {}

  beforeEach(() => {
    mockProps = {
      match: {
        params: {
          category: 'mounts',
          content: 1
        }
      }
    }
  })

  it('renders correctly', () => {
    const component = shallow(<NavBar {...mockProps} />)

    expect(component).toMatchSnapshot()
  })

  it('renders when [match.params.category] prop is null', () => {
    mockProps.match.params.category = null

    const component = shallow(<NavBar {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})
