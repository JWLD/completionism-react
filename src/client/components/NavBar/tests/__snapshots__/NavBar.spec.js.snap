// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    match={
      Object {
        "params": Object {
          "category": "mounts",
          "content": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <LeftWrap>
          <NavLink
            to="/"
          >
            <HomeIcon />
          </NavLink>
          <styled.span>
            mounts
          </styled.span>
        </LeftWrap>,
        <CentreWrap>
          <NavLink
            to="/browse/mounts/9"
          >
            <Styled(FaArrowCircleLeft) />
          </NavLink>
          <MainTitle>
            Mainland
          </MainTitle>
          <NavLink
            to="/browse/mounts/2"
          >
            <Styled(FaArrowCircleRight) />
          </NavLink>
        </CentreWrap>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavLink
              to="/"
            >
              <HomeIcon />
            </NavLink>,
            <styled.span>
              mounts
            </styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <HomeIcon />,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "mounts",
            },
            "ref": null,
            "rendered": "mounts",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavLink
              to="/browse/mounts/9"
            >
              <Styled(FaArrowCircleLeft) />
            </NavLink>,
            <MainTitle>
              Mainland
            </MainTitle>,
            <NavLink
              to="/browse/mounts/2"
            >
              <Styled(FaArrowCircleRight) />
            </NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(FaArrowCircleLeft) />,
              "to": "/browse/mounts/9",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Mainland",
            },
            "ref": null,
            "rendered": "Mainland",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(FaArrowCircleRight) />,
              "to": "/browse/mounts/2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <LeftWrap>
            <NavLink
              to="/"
            >
              <HomeIcon />
            </NavLink>
            <styled.span>
              mounts
            </styled.span>
          </LeftWrap>,
          <CentreWrap>
            <NavLink
              to="/browse/mounts/9"
            >
              <Styled(FaArrowCircleLeft) />
            </NavLink>
            <MainTitle>
              Mainland
            </MainTitle>
            <NavLink
              to="/browse/mounts/2"
            >
              <Styled(FaArrowCircleRight) />
            </NavLink>
          </CentreWrap>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavLink
                to="/"
              >
                <HomeIcon />
              </NavLink>,
              <styled.span>
                mounts
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <HomeIcon />,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "mounts",
              },
              "ref": null,
              "rendered": "mounts",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavLink
                to="/browse/mounts/9"
              >
                <Styled(FaArrowCircleLeft) />
              </NavLink>,
              <MainTitle>
                Mainland
              </MainTitle>,
              <NavLink
                to="/browse/mounts/2"
              >
                <Styled(FaArrowCircleRight) />
              </NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(FaArrowCircleLeft) />,
                "to": "/browse/mounts/9",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Mainland",
              },
              "ref": null,
              "rendered": "Mainland",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(FaArrowCircleRight) />,
                "to": "/browse/mounts/2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<NavBar /> renders when [match.params.category] prop is null 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    match={
      Object {
        "params": Object {
          "category": null,
          "content": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <LeftWrap>
          <NavLink
            to="/"
          >
            <HomeIcon />
          </NavLink>
        </LeftWrap>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavLink
              to="/"
            >
              <HomeIcon />
            </NavLink>,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <HomeIcon />,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <LeftWrap>
            <NavLink
              to="/"
            >
              <HomeIcon />
            </NavLink>
          </LeftWrap>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavLink
                to="/"
              >
                <HomeIcon />
              </NavLink>,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <HomeIcon />,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
